presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisTableDescriptionSupplier.java

```diff
@@ -0,0 +1,124 @@
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class get() method reads the table description file stored in Kinesis directory
+ * and then creates user defined field for Presto Table.
+ */
+public class KinesisTableDescriptionSupplier
+        implements Supplier<Map<SchemaTableName, KinesisStreamDescription>>, ConnectorShutdown
+{
+    private static final Logger log = Logger.get(KinesisTableDescriptionSupplier.class);
+
+    private final KinesisConnectorConfig kinesisConnectorConfig;
+    private final JsonCodec<KinesisStreamDescription> streamDescriptionCodec;
+    private final S3TableConfigClient s3TableConfigClient;
+
+    @Inject
+    KinesisTableDescriptionSupplier(KinesisConnectorConfig kinesisConnectorConfig,
+            JsonCodec<KinesisStreamDescription> streamDescriptionCodec,
+            S3TableConfigClient anS3Client)
+    {
+        this.kinesisConnectorConfig = requireNonNull(kinesisConnectorConfig, "kinesisConnectorConfig is null");
+        this.streamDescriptionCodec = requireNonNull(streamDescriptionCodec, "streamDescriptionCodec is null");
+        this.s3TableConfigClient = requireNonNull(anS3Client, "S3 table config client is null");
+    }
+
+    @Override
+    public Map<SchemaTableName, KinesisStreamDescription> get()
+    {
+        if (this.s3TableConfigClient.isUsingS3()) {
+            return this.s3TableConfigClient.getTablesFromS3();
+        }
+        else {
+            return getTablesFromDirectory();
+        }
+    }
+
+    public Map<SchemaTableName, KinesisStreamDescription> getTablesFromDirectory()
+    {
+        ImmutableMap.Builder<SchemaTableName, KinesisStreamDescription> builder = ImmutableMap.builder();
+        try {
+            for (Path file : listFiles(Paths.get(kinesisConnectorConfig.getTableDescriptionDir()))) {
+                if (Files.isRegularFile(file) && file.getFileName().toString().endsWith("json")) {
+                    KinesisStreamDescription table = streamDescriptionCodec.fromJson(Files.readAllBytes(file));
+                    String schemaName = firstNonNull(table.getSchemaName(), kinesisConnectorConfig.getDefaultSchema());
+                    log.debug("Kinesis table %s %s %s", schemaName, table.getTableName(), table);
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, KinesisStreamDescription> tableDefinitions = builder.build();
+            log.debug("Loaded table definitions: %s", tableDefinitions.keySet());
+
+            return tableDefinitions;
+        }
+        catch (IOException e) {
+            log.warn(e, "Error: ");
+            throw Throwables.propagate(e);
+        }
+    }
+
+    /**
+     * Shutdown any periodic update jobs.
+     */
+    @Override
+    public void shutdown()
```

Comment:
Annotate this method with `@PreDestroy` to have it shutdown automatically when the lifecycle manager is stopped. After this change, we should be able to remove the `ConnectorShutdown` interface as nothing will use it.

Reply:
done